pvoutput/grid_search.py

```diff
@@ -0,0 +1,763 @@
+            improved accuracy. e.g. set to 27700 (OSGB36 / British National Grid) if searching the
+            British Isles.
+
+    Returns:
+        As per `coords` but containing only the subset of the input coordinates which fall within
+        `buffer` km of the bounding box.
+    """
+    if search_radius is None:
+        bounds = buffer_bounding_box_bounds(bbox, buffer, local_crs_epsg)
+        coords["selected"] = (bbox[0] <= coords.latitude <= bbox[2]) & (
+            bbox[1] <= coords.longitude <= bbox[3]
+        )
+    else:
+        bbox_ = buffer_bounding_box(bbox, buffer, local_crs_epsg)[0]
+        coords_ = coords.to_crs(f"EPSG:{local_crs_epsg}").buffer(search_radius * 1000.0)
+        coords["selected"] = coords_.intersects(bbox_)
+    return coords.loc[coords["selected"]].drop(columns="selected")
+
+
+def buffer_bounding_box(
+    bbox: Optional[Iterable[float]] = None, buffer: float = 0, local_crs_epsg: int = 4087
+) -> gpd.GeoSeries:
+    """Buffer a bounding box by a distance in km.
+
+    Args:
+        bbox:
+            Four element iterable defining a bounding box: [min_lat, min_lon, max_lat, max_lon].
+        buffer:
+            Optionally buffer the country boundaries before clipping, in kilometers.
+        local_crs_epsg:
+            Optionally provide the EPSG code of a local Co-ordinate Reference System (CRS) for
+            improved accuracy. e.g. set to 27700 (OSGB36 / British National Grid) if searching the
+            British Isles.
+
+    Returns:
+        A geopandas GeoSeries containing the buffered geometry.
+    """
+    bbox_ = Polygon(
+        [(bbox[1], bbox[0]), (bbox[1], bbox[2]), (bbox[3], bbox[2]), (bbox[3], bbox[0])]
+    )
+    bbox_ = (
+        gpd.GeoSeries(bbox_)
+        .set_crs("EPSG:4326")
+        .to_crs(f"EPSG:{local_crs_epsg}")
+        .buffer(buffer * 1000.0)
+    )
+    return bbox_
+
+
+def buffer_bounding_box_bounds(
+    bbox: Optional[Iterable[float]] = None, buffer: float = 0, local_crs_epsg: int = 4087
+) -> Tuple[float]:
+    """Buffer a bounding box by a distance in km.
+
+    Args:
+        bbox:
+            Four element iterable defining a bounding box: [min_lat, min_lon, max_lat, max_lon].
+        buffer:
+            Optionally buffer the country boundaries before clipping, in kilometers.
+        local_crs_epsg:
+            Optionally provide the EPSG code of a local Co-ordinate Reference System (CRS) for
+            improved accuracy. e.g. set to 27700 (OSGB36 / British National Grid) if searching the
+            British Isles.
+
+    Returns:
+        Four element tuple defining a bounding box: (min_lat, min_lon, max_lat, max_lon).
+    """
+    bbox_ = buffer_bounding_box(bbox, buffer, local_crs_epsg)
+    new_bounds = bbox_.to_crs("EPSG:4326").bounds.loc[0].to_numpy()
+    return new_bounds[1], new_bounds[0], new_bounds[3], new_bounds[2]
+
+
+def bounding_box_from_radius(
```

Comment:
I always get a bit worried when I see a file with > 500 lines. I wonder if some of this functions i.e the clipping ones, could be move to a separate file?

Reply:
500 line limit?!? 🤦‍♂️😂😂 JK, I'll see if it can be broken up sensibly before merging the PR. Tbf, over half of the lines are docstrings.

Reply:
Yea we have no strict limit. So, to be honest, if you are happy with it leave it how it is.